LAKAREBER JULIET LAURA 
B41236 
S26B38-069

#Approaches to reading data
1. We shoukd know the extension of the file e.g xls, xlsx, csv, json etc.
2. We tell where the file is located eg;
- in the same folder with the notebook
- in in another folder/dir

In [13]:
import pandas as pd

## reading the dataset
student_data = pd.read_csv()
student_data

In [1]:
## location: in the same folder is that we write the same file name
student_data = pd.read_csv('StudentsPerformance.csv')
student_data

NameError: name 'pd' is not defined

In [ ]:
student_data.shape

(1000, 8)

In [ ]:
student_data.head() ##

,gender,race/ethnicity,parental level of education,lunch,test preparation course,math score,reading score,writing score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88
2,female,group B,master's degree,standard,none,90,95,93
3,male,group A,associate's degree,free/reduced,none,47,57,44
4,male,group C,some college,standard,none,76,78,75


In [ ]:
student_data.head(10) ##

,gender,race/ethnicity,parental level of education,lunch,test preparation course,math score,reading score,writing score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88
2,female,group B,master's degree,standard,none,90,95,93
3,male,group A,associate's degree,free/reduced,none,47,57,44
4,male,group C,some college,standard,none,76,78,75
5,female,group B,associate's degree,standard,none,71,83,78
6,female,group B,some college,standard,completed,88,95,92
7,male,group B,some college,free/reduced,none,40,43,39
8,male,group D,high school,free/reduced,completed,64,64,67
9,female,group B,high school,free/reduced,none,38,60,50


In [ ]:
student_data.tail() ##

,gender,race/ethnicity,parental level of education,lunch,test preparation course,math score,reading score,writing score
995,female,group E,master's degree,standard,completed,88,99,95
996,male,group C,high school,free/reduced,none,62,55,55
997,female,group C,high school,free/reduced,completed,59,71,65
998,female,group D,some college,standard,completed,68,78,77
999,female,group D,some college,free/reduced,none,77,86,86


In [ ]:
student_data.columns

Index(['gender', 'race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course', 'math score', 'reading score',
       'writing score'],
      dtype='str')

In [ ]:
student_data.describe()

,math score,reading score,writing score
count,1000.00000,1000.000000,1000.000000
mean,66.08900,69.169000,68.054000
std,15.16308,14.600192,15.195657
min,0.00000,17.000000,10.000000
25%,57.00000,59.000000,57.750000
50%,66.00000,70.000000,69.000000
75%,77.00000,79.000000,79.000000
max,100.00000,100.000000,100.000000


In [ ]:
student_data.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   gender                       1000 non-null   str  
 1   race/ethnicity               1000 non-null   str  
 2   parental level of education  1000 non-null   str  
 3   lunch                        1000 non-null   str  
 4   test preparation course      1000 non-null   str  
 5   math score                   1000 non-null   int64
 6   reading score                1000 non-null   int64
 7   writing score                1000 non-null   int64
dtypes: int64(3), str(5)
memory usage: 62.6 KB


1. Find the mean math score?
2. what is the median reading score?
3. What is the modal writing score?
4. Find the average maths score for females?
5. Construct a new column called 'Total (summation of math reading and writing ascore) and find its average

1. Mean math score.

In [ ]:
mean_math_score = student_data['math score'].mean() ## mean of math score
mean_math_score

np.float64(66.089)

The mean math score of math is 66.089

2. Median reading score.

In [ ]:
median_reading_score = student_data['reading score'].median() ## median of reading score
median_reading_score

np.float64(70.0)

The median of reading score is 70.0

3. Modal writing score.

In [ ]:
modal_writing_score = student_data['writing score'].mode() ## mode of writing score
modal_writing_score

0    74
Name: writing score, dtype: int64

The modal wrting score is 74

4. average maths score for females.

In [ ]:
female_student = student_data[student_data['gender']=='female']
female_student['math score'].mean()

np.float64(63.633204633204635)

- From the results the average for female students is 63.6332.

5. Total_marks for reading, math and writing score.

In [ ]:
student_data['total marks']= student_data['math score'] + student_data['reading score'] + student_data['writing score']
student_data['total marks']
student_data

,gender,race/ethnicity,parental level of education,lunch,test preparation course,math score,reading score,writing score,total score,total marks
0,female,group B,bachelor's degree,standard,none,72,72,74,218,218
1,female,group C,some college,standard,completed,69,90,88,247,247
2,female,group B,master's degree,standard,none,90,95,93,278,278
3,male,group A,associate's degree,free/reduced,none,47,57,44,148,148
4,male,group C,some college,standard,none,76,78,75,229,229
...,...,...,...,...,...,...,...,...,...,...
995,female,group E,master's degree,standard,completed,88,99,95,282,282
996,male,group C,high school,free/reduced,none,62,55,55,172,172
997,female,group C,high school,free/reduced,completed,59,71,65,195,195
998,female,group D,some college,standard,completed,68,78,77,223,223


In [ ]:
student_data['total marks'].mean()

np.float64(203.312)

- From the results above the mean total score is 203.312.

Average of the total_marks.

In [ ]:
Average = Total_students.mean() 
Average

np.float64(203.312)

The average of the total_marks is 203.312.

Measures Of Position.

[[1/4], [2/4], [3/4], [4/4]]
[[0.25], [0.5], [0.75], [1.0]]

In [ ]:
lower_quartile = student_data['math score'].quantile(0.25) ## lower quartile
lower_quartile

np.float64(57.0)

In [ ]:

upper_quartile = student_data['math score'].quantile(0.75) ## upper quartile

In [ ]:
import numpy as np

ninty_per = np.percentile(student_data['math score'], 90) ##  90th percentile
ninty_per

np.float64(86.0)

student_data = pd.read_xlsx()

In [8]:
## location: outside the folder
student_data = pd.read_excel("C:\\Users\\LENOVO\\OneDrive\\Documents\\StudentsPerformance.xlsx")
student_data

,gender,race/ethnicity,parental level of education,lunch,test preparation course,math score,reading score,writing score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88
2,female,group B,master's degree,standard,none,90,95,93
3,male,group A,associate's degree,free/reduced,none,47,57,44
4,male,group C,some college,standard,none,76,78,75
...,...,...,...,...,...,...,...,...
995,female,group E,master's degree,standard,completed,88,99,95
996,male,group C,high school,free/reduced,none,62,55,55
997,female,group C,high school,free/reduced,completed,59,71,65
998,female,group D,some college,standard,completed,68,78,77


[[1/4],[2/4],[3/4],[4/4]] [[0.25],[0.5],[0.75],[1.0]]

In [ ]:
inter_quartile = student_data['math score'].quantile (0.75) - student_data['math score'].quantile (0.25)
inter_quartile

student_data = pd.read_xlsx()

In [14]:
import pandas as pd
## location: in the same folder
student_data = pd.read_excel("C:\\Users\\LENOVO\\OneDrive\\Desktop\\PROBABILITY\\daddy.xlsx")
student_data

,number_courses,time_study,Marks
0,3,4.508,19.202
1,4,0.096,7.734
2,4,3.133,13.811
3,6,7.909,53.018
4,8,7.811,55.299
...,...,...,...
95,6,3.561,19.128
96,3,0.301,5.609
97,4,7.163,41.444
98,7,0.309,12.027
